src/Microsoft.OData.Core/ODataBatchReaderStream.cs

```diff
@@ -8,332 +8,68 @@ namespace Microsoft.OData
-        /// <param name="batchBoundary">The boundary string for the batch structure itself.</param>
         /// <param name="batchEncoding">The encoding to use to read from the batch stream.</param>
         internal ODataBatchReaderStream(
-            ODataRawInputContext inputContext,
-            string batchBoundary,
             Encoding batchEncoding)
         {
-            Debug.Assert(inputContext != null, "inputContext != null");
-            Debug.Assert(!string.IsNullOrEmpty(batchBoundary), "!string.IsNullOrEmpty(batchBoundary)");
-
-            this.inputContext = inputContext;
-            this.batchBoundary = batchBoundary;
             this.batchEncoding = batchEncoding;
 
-            this.batchBuffer = new ODataBatchReaderStreamBuffer();
-
-            // When we allocate a batch reader stream we will in almost all cases also call ReadLine
-            // (to read the headers of the parts); so allocating it here.
-            this.lineBuffer = new byte[LineBufferLength];
-
-            this.mediaTypeResolver = ODataMediaTypeResolver.GetMediaTypeResolver(inputContext.Container);
-        }
-
-        /// <summary>
-        /// The boundary string for the batch structure itself.
-        /// </summary>
-        internal string BatchBoundary
-        {
-            get
-            {
-                return this.batchBoundary;
-            }
-        }
-
-        /// <summary>
-        /// The boundary string for the current changeset (only set when reading a changeset
-        /// or an operation in a changeset).
-        /// </summary>
-        /// <remarks>When not reading a changeset (or operation in a changeset) this field is null.</remarks>
-        internal string ChangeSetBoundary
-        {
-            get
-            {
-                return this.changesetBoundary;
-            }
-        }
-
-        /// <summary>
-        /// The current boundary string to be used for reading with delimiter.
-        /// </summary>
-        /// <remarks>This is the changeset boundary when reading a changeset or the batch boundary otherwise.</remarks>
-        private IEnumerable<string> CurrentBoundaries
-        {
-            get
-            {
-                if (this.changesetBoundary != null)
-                {
-                    yield return this.changesetBoundary;
-                }
-
-                yield return this.batchBoundary;
-            }
+            this.BatchBuffer = new ODataBatchReaderStreamBuffer();
         }
 
         /// <summary>
         /// The current encoding to use when reading from the stream.
         /// </summary>
         /// <remarks>This is the changeset encoding when reading a changeset or the batch encoding otherwise.</remarks>
-        private Encoding CurrentEncoding
+        protected Encoding CurrentEncoding
         {
             get
             {
                 return this.changesetEncoding ?? this.batchEncoding;
             }
         }

```

Comment:
Can move this to multipartmixed.

Reply:
moved.

---
In reply to: [155380026](https://github.com/OData/odata.net/pull/992#discussion_r155380026) [](ancestors = 155380026)